src/dune_cache/local.ml

```diff
@@ -98,15 +67,18 @@ module Artifacts = struct
      If any of the targets couldn't be stored in the temporary directory, then
      the result is [Error] with the corresponding exception. Otherwise, the
      result is [Ok ()]. *)
-  let store_targets_to ~temp_dir ~targets ~mode : unit Or_exn.t =
+  let store_targets_to ~temp_dir ~targets ~(mode : Dune_cache_storage.Mode.t)
+    : unit Or_exn.t
+    =
+    let copy =
+      match mode with
+      | Hardlink -> fun ~src ~dst -> Io.portable_hardlink ~src ~dst
```

Comment:
I inlined this function from `util` because it was the only place where it was used.